src/firewall/iptables.rs

```diff
@@ -231,13 +304,95 @@ impl firewall::FirewallDriver for IptablesDriver {
         Result::Ok(())
     }
 
-    fn teardown_port_forward(
-        &self,
-        _container_id: &str,
-        _port_mappings: Vec<types::PortMapping>,
-        _container_ip: &str,
-    ) -> Result<(), Box<dyn Error>> {
-        todo!();
+    fn teardown_port_forward(&self, tear: TeardownPortForward) -> Result<(), Box<dyn Error>> {
```

Comment:
Why does teardown_port_forward accept a struct but setup_port_forward doe snot. Can you please make this consistent.

Reply:
I wanted to defer those changes ... would you be comfy if we merge this and then I do that immediately? This PR is already big enough and sweeping changes.

Reply:
sure